Foundation/NSTask.h

```diff
@@ -0,0 +1,87 @@
+/*	NSTask.h
+	Copyright (c) 1996-2019, Apple Inc. All rights reserved.
+*/
+
+#import <Foundation/NSObject.h>
+#import <Foundation/NSNotification.h>
+
+@class NSArray<ObjectType>, NSDictionary<KeyType, ObjectType>, NSString;
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSInteger, NSTaskTerminationReason) {
+    NSTaskTerminationReasonExit = 1,
+    NSTaskTerminationReasonUncaughtSignal = 2
+} API_AVAILABLE(macos(10.6)) API_UNAVAILABLE(ios, watchos, tvos);
+
+@interface NSTask : NSObject
+
+// Create an NSTask which can be run at a later time
+// An NSTask can only be run once. Subsequent attempts to
+// run an NSTask will raise.
+// Upon task death a notification will be sent
+//   { Name = NSTaskDidTerminateNotification; object = task; }
+//
+
+- (instancetype)init NS_DESIGNATED_INITIALIZER;
+
+// these methods can only be set before a launch
+@property (nullable, copy) NSURL *executableURL API_AVAILABLE(macos(10.13)) API_UNAVAILABLE(ios, watchos, tvos);
+@property (nullable, copy) NSArray<NSString *> *arguments;
+@property (nullable, copy) NSDictionary<NSString *, NSString *> *environment; // if not set, use current
+@property (nullable, copy) NSURL *currentDirectoryURL API_AVAILABLE(macos(10.13)) API_UNAVAILABLE(ios, watchos, tvos);
```

Comment:
should these availability attributes be updated?
```suggestion
@property (nullable, copy) NSURL *currentDirectoryURL API_AVAILABLE(macos(10.13));
```

Reply:
the "became" available attributes look valid to me. should they be kept in?

Reply:
Do we want the headers to be compatible with macOS too?

Reply:
It'd be a nice to have but not a requirement, I think